pkg/client/add_events_test.go

```diff
@@ -83,27 +84,29 @@ func extract(req *http.Request) (add_events.AddEventsRequest, error) {
 }
 
 func (s *SuiteAddEvents) TestAddEventsRetry(assert, require *td.T) {
-	attempt := 0
-	wasSuccesful := false
+	attempt := atomic.Int32{}
```

Comment:
I assume we need to use either an atomic variable or lock here when updating attempts since this function can be called by different go routines possibly running on different threads concurrently, right?

Reply:
Yes. Majority of these races were caused by these helper variables. I was setting them in the callback and reading them in the main code.